python-backend/app/api/mines/mine/resources/mine.py

```diff
@@ -197,46 +217,42 @@ def mine_operation_code_processor(self, mine_status, index):
         except IndexError:
             return None
 
-    def mine_status_processor(self, mine_status, mine_guid):
+    def mine_status_processor(self, mine_status, mine):
+        if not mine_status:
+            return mine.mine_status
         try:
             mine_status_xref = MineStatusXref.find_by_codes(
                 self.mine_operation_code_processor(mine_status, 0),
                 self.mine_operation_code_processor(mine_status, 1),
                 self.mine_operation_code_processor(mine_status, 2))
             if not mine_status_xref:
-                self.raise_error(
-                    400,
-                    'Error: Invalid status_code, reason_code, and sub_reason_code combination., '
-                ), 400
+                raise BadRequest(
+                    'Invalid status_code, reason_code, and sub_reason_code combination.')
 
-            existing_status = MineStatus.find_by_mine_guid(mine_guid)
+            existing_status = mine.mine_status
```

Comment:
Are you sure this works? Curious why previously we had to find it.

Reply:
This function was accepting the mine_guid. I changed it to accept the mine object so i could assign the relationship so don't need to save the mine before saving the status.

It works.